RecoLocalCalo/HGCalRecAlgos/interface/HGCalRecHitAbsAlgo.h

```diff
@@ -0,0 +1,32 @@
+#ifndef RecoLocalCalo_HGCalRecAlgos_HGCalRecHitAbsAlgo_HH
+#define RecoLocalCalo_HGCalRecAlgos_HGCalRecHitAbsAlgo_HH
+
+/** \class HGCalRecHitAbsAlgo
+  *  Template algorithm to make rechits from uncalibrated rechits
+  *
+  *
+  *
+  *  \author Valeri Andreev
+  */
+
+#include <vector>
+#include "DataFormats/HGCRecHit/interface/HGCRecHit.h"
+#include "DataFormats/HGCRecHit/interface/HGCUncalibratedRecHit.h"
+
+class HGCalRecHitAbsAlgo
+{
+ public:
+
+  /// Constructor
+  //HGCalRecHitAbsAlgo() { };
+
+  /// Destructor
+  virtual ~HGCalRecHitAbsAlgo() { };
+
+  /// make rechits from dataframes
+
+  virtual void setADCToGeVConstant(const float& value) = 0;
```

Comment:
basic types should not be passed by reference. It generates more code than passing by value. (keeping const is OK)